storage/api/Storage/Storage.cs

```diff
@@ -705,6 +707,43 @@ private void GenerateSignedUrl(string bucketName, string objectName)
         }
         // [END storage_generate_signed_url]
 
+        // [START storage_generate_signed_url_v4]
+        private void GenerateV4SignedGetUrl(string bucketName, string objectName)
+        {
+            UrlSigner urlSigner = UrlSigner
+            .FromServiceAccountPath(Environment.GetEnvironmentVariable("GOOGLE_APPLICATION_CREDENTIALS"))
+            .WithSigningVersion(SigningVersion.V4);
+            string url =
+                urlSigner.Sign(bucketName, objectName, TimeSpan.FromHours(1), HttpMethod.Get);
+            Console.WriteLine("Generated GET signed URL:");
+            Console.WriteLine(url);
+            Console.WriteLine("You can use this URL with any user agent, for example:");
+            Console.WriteLine("curl '" + url + "'");

```

Comment:
I'd personally use an interpolated string literal here: `Console.WriteLine($"curl '{url}'");` - and likewise at line 742. But it's not a big deal.

Reply:
+1